vaadin-grid-flow-parent/vaadin-grid-testbench/src/main/java/com/vaadin/flow/component/grid/testbench/GridElement.java

```diff
@@ -430,4 +430,77 @@ public void clickSelectAll() {
                 .id("selectAllCheckbox");
         selectAllCheckbox.click();
     }
+
+    /**
+     * Gets the rows (present in the DOM) specified by the lower and upper row
+     * indexes.
+     *
+     * @param firstRowIndex
+     *            the lower row index to be retrieved (inclusive)
+     * @param lastRowIndex
+     *            the upper row index to be retrieved (inclusive)
+     * @return a {@link GridTRElement} list with the rows contained between the
+     *         given coordinates.
+     * @throws IndexOutOfBoundsException
+     *             if either of the provided row indexes do not exist
+     */
+    public List<GridTRElement> getRows(int firstRowIndex, int lastRowIndex)
+            throws IndexOutOfBoundsException {
+        int rowCount = getRowCount();
+        if (firstRowIndex < 0 || lastRowIndex < 0 || firstRowIndex >= rowCount
+                || lastRowIndex >= rowCount) {
+            throw new IndexOutOfBoundsException(
+                    "firstRowIndex and lastRowIndex: expected to be 0.."
+                            + (rowCount - 1) + " but were " + firstRowIndex
+                            + " and " + lastRowIndex);
+        }
+        String script = "var grid = arguments[0];"
+                + "var firstRowIndex = arguments[1];"
+                + "var lastRowIndex = arguments[2];"
+                + "var rowsInDom = grid.$.items.children;"
+                + "return Array.from(rowsInDom).filter((row) => { return row.index >= firstRowIndex && row.index <= lastRowIndex;});";
+        return ((ArrayList<TestBenchElement>) executeScript(script, this,
+                firstRowIndex, lastRowIndex)).stream()
+                .map(elem -> elem.wrap(GridTRElement.class)).toList();
+    }
+
+    /**
+     * Gets all the currently visible rows.
+     *
+     * @return a {@link GridTRElement} list representing the currently visible
+     *         rows.
+     */
+    public List<GridTRElement> getVisibleRows() {
+        return getRows(getFirstVisibleRowIndex(), getLastVisibleRowIndex());
+    }
+
+    /**
+     * Gets the grid cells for the given row and column elements.
+     *
+     * @param rowIndex
+     *            the row index
+     * @param columnElements
+     *            the column elements
+     * @return a {@link GridTHTDElement} list with the cells for the given
+     *         coordinates.
+     */
+    public List<GridTHTDElement> getCells(int rowIndex,
+            GridColumnElement... columnElements) {
+        GridTRElement row = getRow(rowIndex);
+        return row.getCells(columnElements);
```

Comment:
This throws an NPE if the row is `null` / not in the DOM (I disabled `allRowsVisible` to test):

<img width="462" alt="Screenshot 2023-10-10 at 14 20 43" src="https://github.com/vaadin/flow-components/assets/1222264/e5d11bd4-6098-4aa6-b90d-7518cdac2080">